### ISSUE TITLE
SIP NAPTR connections fail when TURN NAPTR exists

### ISSUE THREAD

**Issue by dpocock** [jitsi/jitsi#278]
This was observed with Jitsi 2.9.5478

My DNS had the following two NAPTR records for ws.sip5060.net:

```
ws.sip5060.net.		524	IN	NAPTR	10 0 "s" "SIPS+D2T" "" _sips._tcp.ws.sip5060.net.
ws.sip5060.net.		524	IN	NAPTR	10 0 "s" "RELAY:turn.udp" "" _turn._udp.sip5060.net.
```

I would add a SIP account "user@ws.sip5060.net" and Jitsi would complain that it couldn't connect.

The following exception appeared in the logs:

```
java.lang.NullPointerException
        at net.java.sip.communicator.util.NetworkUtils.getProtocolPriority(NetworkUtils.java:844)
        at net.java.sip.communicator.util.NetworkUtils.access$000(NetworkUtils.java:38)
        at net.java.sip.communicator.util.NetworkUtils$1.compare(NetworkUtils.java:802)
        at net.java.sip.communicator.util.NetworkUtils$1.compare(NetworkUtils.java:779)
        at java.util.TimSort.countRunAndMakeAscending(TimSort.java:324)
        at java.util.TimSort.sort(TimSort.java:189)
        at java.util.TimSort.sort(TimSort.java:173)
        at java.util.Arrays.sort(Arrays.java:659)
        at net.java.sip.communicator.util.NetworkUtils.getNAPTRRecords(NetworkUtils.java:778)
        at net.java.sip.communicator.impl.protocol.sip.net.AutoProxyConnection$LocalNetworkUtils.getNAPTRRecords(AutoProxyConnection.java:72)
        at net.java.sip.communicator.impl.protocol.sip.net.AutoProxyConnection.getNextAddressInternal(AutoProxyConnection.java:233)
        at net.java.sip.communicator.impl.protocol.sip.net.AutoProxyConnection.getNextAddressFromDns(AutoProxyConnection.java:184)
        at net.java.sip.communicator.impl.protocol.sip.net.AutoProxyConnection.getNextAddressInternal(AutoProxyConnection.java:208)
        at net.java.sip.communicator.impl.protocol.sip.net.AutoProxyConnection.getNextAddressFromDns(AutoProxyConnection.java:184)
        at net.java.sip.communicator.impl.protocol.sip.net.ProxyConnection.getNextAddress(ProxyConnection.java:130)
        at net.java.sip.communicator.impl.protocol.sip.ProtocolProviderServiceSipImpl.registerUsingNextAddress(ProtocolProviderServiceSipImpl.java:2647)
        at net.java.sip.communicator.impl.protocol.sip.ProtocolProviderServiceSipImpl.register(ProtocolProviderServiceSipImpl.java:371)
        at net.java.sip.communicator.util.account.LoginManager$RegisterProvider.run(LoginManager.java:434)
2016-06-28 16:07:13.721 INFO: [484] impl.protocol.sip.SipLogger.logInfo().196 Info from the JAIN-SIP stack: the sip stack timer gov.nist.javax.sip.stack.timers.DefaultSipTimer has been stopped
2016-06-28 16:07:14.722 INFO: [484] impl.protocol.sip.SipLogger.logInfo().196 Info from the JAIN-SIP stack: the sip stack timer gov.nist.javax.sip.stack.timers.DefaultSipTimer has been stopped
2016-06-28 16:07:15.723 SEVERE: [484] impl.protocol.sip.ProtocolProviderServiceSipImpl.register().373 No address found for ProtocolProviderServiceSipImpl(user@ws.sip5060.net (SIP))
```

I removed the NAPTR for "RELAY:turn.udp" from the DNS zone, restarted Jitsi and now it works correctly with that domain.

**Comment by damencho** [comment]
I believe @ibauersachs fixed a similar issue with: https://github.com/jitsi/jitsi/commit/3ccebdc455b39218f42243d35b4d12e3fec7e822
And this is available in Jitsi build 2.9.5522

**Comment by ibauersachs** [comment]
Not similar, exactly this.
